src/ServiceControl.Config/UI/License/LicenseComponentFactory.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace ServiceControl.Config.UI.License
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using ServiceControlInstaller.Engine.LicenseMgmt;
+
+    class LicenseComponentFactory
+    {
+        readonly DateTime today;
+        const int ExpiryWarningPeriodInDays = 10;
+
+        const string TrialLicenseText = "Please extend your trial or purchase a license to continue using the Particular Service Platform.";
+        const string UpgradeProtectionLicenseText = "Please extend your upgrade protection so that we can continue to provide you with support and new versions of the Particular Service Platform.";
+        const string SubscriptionLicenseText = "Please extend your license to continue using the Particular Service Platform.";
+
+        CountInflector daysInflector = new CountInflector
+        {
+            Singular = "{0} day",
+            Plural = "{0} days"
+        };
+
+        public LicenseComponentFactory(DateTime? today = null)
+        {
+            this.today = today ?? DateTime.Today;
+        }
+
+        public IEnumerable<LicenseComponent> CreateComponents(LicenseDetails details)
+        {
+            if (details.IsTrialLicense)
+            {
+                yield return new LicenseComponent
+                {
+                    Label = "Platform license type:",
+                    Value = "Trial"
+                };
+
+                if (details.ExpirationDate.HasValue)
+                {
+                    yield return TrialExpiryComponent(details.ExpirationDate.Value);
+                }
+            }
+            else
+            {
+                yield return new LicenseComponent
+                {
+                    Label = "Platform license type:",
+                    Value = $"{details.LicenseType}, {details.Edition}"
+                };
+
+                if (details.ExpirationDate.HasValue)
+                {
+                    yield return PlatformExpiryComponent(details.ExpirationDate.Value);
+                }
+
+                if (details.UpgradeProtectionExpiration.HasValue)
```

Comment:
Shouldn't this be before expiration date? All of the perpetual licenses I have seen have both.

Reply:
They have both in the file but we don't load expiration date for perpetual licenses

Reply:
And because these are doing a `yield return`, it won't short circuit anyway. If your license reports having both, we will display both.